lib/spack/spack/build_systems/intel.py

```diff
@@ -818,11 +817,11 @@ def scalapack_libs(self):
         elif ('^mpich@2:' in spec_root or
               '^mvapich2' in spec_root or
               '^intel-mpi' in spec_root or
+              '^intel-mpi-external' in spec_root or
               '^intel-parallel-studio' in spec_root):
             blacs_lib = 'libmkl_blacs_intelmpi'
         elif ('^mpt' in spec_root or
-              '^hpe-mpi' in spec_root or
-              '^intel-mpi-external' in spec_root):
```

Comment:
having intel-mpi in hpe-mpi section was a bug (but not affecting as it was for external package in Juelich)